### DISCUSSION TITLE
How to get the Dns Time from the ligthhouse reports.??

### DISCUSSION THREAD

**Discussion by ashwinrajanta** [GoogleChrome/lighthouse#15644] [Q&A] [answered]
Does google lighthouse report the dns timings in the lighthouse report? if so how to retrieve the values?? 
Note. Running the lighthouse via cli inside a docker

**Comment by ashwinrajanta** [comment]
if not directly also. can it derived based on some calculations?

**Comment by adamraine** [comment]
DNS timings are not included in the report

**Comment by ashwinrajanta** [comment]
[defaultPass.devtoolslog.json](https://github.com/GoogleChrome/lighthouse/files/13519606/defaultPass.devtoolslog.json)


@adamraine thanks for responding..   But after execution there is a directory created called "latest-run" and inside that few files are created. in one of these files if you see i could see "Network.responseReceived", there i could see something called dnsStart dnsEnd params

What is that.? i have attached the sample json file


`{"method":"Network.responseReceived","params":{"requestId":"79CAF8387324785A8E2CFE5289FFA15E","loaderId":"79CAF8387324785A8E2CFE5289FFA15E","timestamp":379.08508,"type":"Document","response":{"url":"https://www.facebook.com/","status":200,"statusText":"OK","headers":{"Cache-Control":"private, no-cache, no-store, must-revalidate","Connection":"keep-alive","Content-Encoding":"br","Content-Type":"text/html; charset=\"utf-8\"","Date":"Fri, 06 Oct 2023 00:42:35 GMT","Expires":"Sat, 01 Jan 2000 00:00:00 GMT","Pragma":"no-cache","Strict-Transport-Security":"max-age=15552000; preload","Transfer-Encoding":"chunked","Vary":"Accept-Encoding","X-Content-Type-Options":"nosniff","X-FB-Debug":"Nyn/HxSuTNPan0hhH63/K0m0hydokNX3PiAsEKt1QNSCsW7QR0cccThYW0h7/PN5oGRLn9+xEboIxTvzlhW2Eg==","X-Frame-Options":"DENY","X-XSS-Protection":"0","content-security-policy":"default-src data: blob: 'self' https://*.fbsbx.com 'unsafe-inline' *.facebook.com *.fbcdn.net 'unsafe-eval';script-src *.facebook.com *.fbcdn.net *.facebook.net *.google-analytics.com *.google.com 127.0.0.1:* 'unsafe-inline' blob: data: 'self' connect.facebook.net 'unsafe-eval';style-src fonts.googleapis.com *.fbcdn.net data: *.facebook.com 'unsafe-inline';connect-src *.facebook.com facebook.com *.fbcdn.net *.facebook.net wss://*.facebook.com:* wss://*.whatsapp.com:* wss://*.fbcdn.net attachment.fbsbx.com ws://localhost:* blob: *.cdninstagram.com 'self' http://localhost:3103 wss://gateway.facebook.com wss://edge-chat.facebook.com wss://snaptu-d.facebook.com wss://kaios-d.facebook.com/ v.whatsapp.net *.fbsbx.com *.fb.com;font-src data: *.gstatic.com *.facebook.com *.fbcdn.net *.fbsbx.com;img-src *.fbcdn.net *.facebook.com data: https://*.fbsbx.com *.tenor.co media.tenor.com facebook.com *.cdninstagram.com fbsbx.com fbcdn.net *.giphy.com connect.facebook.net *.carriersignal.info blob: android-webview-video-poster: googleads.g.doubleclick.net www.googleadservices.com *.whatsapp.net *.fb.com *.oculuscdn.com;media-src *.cdninstagram.com blob: *.fbcdn.net *.fbsbx.com www.facebook.com *.facebook.com https://*.giphy.com data:;frame-src *.doubleclick.net *.google.com *.facebook.com www.googleadservices.com *.fbsbx.com fbsbx.com data: www.instagram.com *.fbcdn.net https://paywithmybank.com https://sandbox.paywithmybank.com;worker-src blob: *.facebook.com data:;block-all-mixed-content;upgrade-insecure-requests;","cross-origin-opener-policy":"unsafe-none","cross-origin-resource-policy":"cross-origin","document-policy":"force-load-at-top","nel":"{\"report_to\":\"network-errors\",\"max_age\":3600,\"failure_fraction\":0.01}","origin-agent-cluster":"?0","permissions-policy":"accelerometer=(), ambient-light-sensor=(), bluetooth=(), camera=(self), geolocation=(self), gyroscope=(), hid=(), idle-detection=(), magnetometer=(), microphone=(self), midi=(), payment=(), screen-wake-lock=(), serial=(), usb=()","report-to":"{\"max_age\":259200,\"endpoints\":[{\"url\":\"https:\\/\\/www.facebook.com\\/ajax\\/browser_error_reports\\/?device_level=unknown\"}]}, {\"max_age\":3600,\"endpoints\":[{\"url\":\"https:\\/\\/www.facebook.com\\/ajax\\/browser_error_reports\\/?device_level=unknown\"}],\"group\":\"network-errors\"}"},"mimeType":"text/html","connectionReused":false,"connectionId":13,"remoteIPAddress":"157.240.198.35","remotePort":443,"fromDiskCache":false,"fromServiceWorker":false,"fromPrefetchCache":false,"encodedDataLength":3204,"timing":{"requestTime":378.622949,"proxyStart":-1,"proxyEnd":-1,"dnsStart":18.463,"dnsEnd":50.546,"connectStart":50.546,"connectEnd":114.922,"sslStart":57.612,"sslEnd":114.903,"workerStart":-1,"workerReady":-1,"workerFetchStart":-1,"workerRespondWithSettled":-1,"sendStart":115.015,"sendEnd":115.193,"pushStart":0,"pushEnd":0,"receiveHeadersStart":459.774,"receiveHeadersEnd":460.543},"responseTime":1696552955713.872,"protocol":"http/1.1","alternateProtocolUsage":"unspecifiedReason","securityState":"insecure","securityDetails":{"protocol":"TLS 1.3","keyExchange":"","keyExchangeGroup":"X25519","cipher":"AES_256_GCM","certificateId":0,"subjectName":"*.facebook.com","sanList":["*.facebook.com","facebook.com"],"issuer":"ca.intelsat.goskope.com","validFrom":1693112699,"validTo":1727240699,"signedCertificateTimestampList":[],"certificateTransparencyCompliance":"unknown","serverSignatureAlgorithm":2052,"encryptedClientHello":false}},"hasExtraInfo":true,"frameId":"8A2E11E91542C395A8A433D685E752D2"},"targetType":"page","sessionId":"D6FFEDDCD6744C8F9B3DDC66642B5EE2"},`

**Reply by benschwarz** [reply, marked as answer]
@ashwinrajanta As @adamraine said in a previous reply, Lighthouse does not include DNS timing information. Timing information is available in the remote debugging API (as you've demonstrated above). 

However, Lighthouse by default loads pages at _full network speed_, and then applies _network simulation_ in order to calculate metrics and network timing information. While you *can access* the timing information from the API (as above), it will represent the unthrottled load, and not the simulated load time. 

That's why there's no DNS timings in the report, and afaik why there won't be in the near future.

**Comment by mbq-shanmukh-jerripothlla** [comment]
Hi @benschwarz 

I have a similar query. In the DevTools Performance tab, we can see detailed phases of network calls. I want to capture these details as well. Do you have any ideas on how this can be done?

The specific metrics I'm looking to capture are:

Total duration
Queuing & connection time
Request waiting time
Content download time
Main thread waiting time
